inc/Data/DataSourceRepository/DataSourceRepositoryInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Data;
+
+use RemoteDataBlocks\Config\Datasource\DatasourceInterface;
+
+defined( 'ABSPATH' ) || exit();
+
+interface DataSourceRepositoryInterface {
```

Comment:
Inject this instead of a concrete `DataSourceRepository` into all the things that need it. Controllers, whatever. Then in testing you just pass a test class that implements the interface and you're golden.